### PR TITLE
moved conditional_probability_alive_matrix function + possible bug fix

### PR DESCRIPTION
**Pull request by erikvdp**
Hi, I've moved the conditional_probability_alive_matrix function to the model. While doing so, I noticed that the matrix doesn't calculate all values, so I've added a row and column. It should be like that no?

### CODE REVIEW

File: lifetimes/plotting.py [hunk 1]
```diff
@@ -73,9 +73,9 @@ def plot_frequency_recency_matrix(model, max_x=None, max_t=None, **kwargs):
         max_t = int(model.data['T'].max())
 
     t = 1  # one unit of time
-    Z = np.zeros((max_t, max_x))
```
Comment (CamDavidsonPilon) [inline lifetimes/plotting.py]: This function can now use the `conditional_probability_alive_matrix` you defined above. Want to use that here?

Comment (CamDavidsonPilon) [inline lifetimes/plotting.py]: oops, nvm. Please ignore this.

File: lifetimes/plotting.py [hunk 2]
```diff
@@ -104,26 +105,17 @@ def plot_probability_alive_matrix(model, max_x=None, max_t=None, **kwargs):
...
-        max_x = int(model.data['frequency'].max())
-
-    if max_t is None:
-        max_t = int(model.data['T'].max())
-
-    Z = np.zeros((max_t, max_x))
-    for i, t_x in enumerate(np.arange(max_t)):
-        for j, x in enumerate(np.arange(max_x)):
-            Z[i, j] = model.conditional_probability_alive(x, t_x, max_t)
+    z = model.conditional_probability_alive_matrix(max_x, max_t)
```
Comment (CamDavidsonPilon) [inline lifetimes/plotting.py]: :+1:

**Comment by CamDavidsonPilon** [comment]
You're correct about the `ranges`, nice catch. Circle keeps failing, but it's unrelated to these tests. Actually, can you add a test for the `conditional_probability_alive_matrix` too please?

**Comment by erikvdp** [comment]
Hi again,
I wanted to check if this test is good enough for you.
I was planning to compare the entries of the <code>conditional_probability_alive_matrix</code> against the corresponding <code>conditional_probability_alive</code> functions, but they seem to be slightly off. Do you have any idea why that is?

**Comment by CamDavidsonPilon** [comment]
My best guess is that it's an off by one error - can you reproduce the inconsistency?

**Comment by erikvdp** [comment]
Hi, The small error in the <code> test_conditional_probability_alive_matrix </code> method was due to the T_max not properly rounded. It should be ok now.

**Comment by CamDavidsonPilon** [comment]
Great, thanks for this contribution! Merging now.

**Pull request merged**